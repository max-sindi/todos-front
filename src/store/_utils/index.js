export getFormData from './getFormData'
export requestResultAction from './requestResultAction'
export handlingRequestResult from './handlingRequestResult'
export createAction from './createAction'
export createFilterValueToSend from './createFilterValueToSend'
