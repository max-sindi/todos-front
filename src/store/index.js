import {compose, applyMiddleware, createStore} from "redux"
import thunk from "redux-thunk"
import {createLogger} from "redux-logger"
import createSaga from 'redux-saga'
import rootSaga from './saga'
import combinedReducers from "./reducers"

const logger = createLogger({diff: true, collapsed: true})
const saga = createSaga()
const reduxDevtools = window.__REDUX_DEVTOOLS_EXTENSION__

const store = createStore(
  combinedReducers,
  undefined,
  compose(
    applyMiddleware( thunk, saga, logger),
    reduxDevtools ? reduxDevtools() : f => f
  )
)

saga.run(rootSaga)

export default store