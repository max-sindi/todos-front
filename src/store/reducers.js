import {combineReducers} from "redux"
import {todos, todo}from "./todos/todosReducer"
import notifications from "./notifications/notificationsReducer"

const reducers = combineReducers({
  todos,
  todo,
  notifications,
})

export default reducers