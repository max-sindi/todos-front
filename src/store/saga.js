import { createRequestInstance, watchRequests } from 'redux-saga-requests'
import { createDriver } from 'redux-saga-requests-axios'
import axios from 'axios'

axios.defaults.baseURL = 'http://localhost:8000'
// axios.defaults.headers['Access-Control-Allow-Origin'] = '*'
// axios.defaults.crossdomain = true

export default function* rootSaga() {
  yield createRequestInstance({ driver: createDriver(axios) })
  yield watchRequests()
}