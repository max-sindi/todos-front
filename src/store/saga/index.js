import { put, all, takeLatest } from 'redux-saga/effects'
import { TodosApi } from 'api'
import { getDataFromResult } from '../todos/actions'
import { defaultErrorCatcher } from '../notifications/actions'
import { createAction } from '../_utils/'

import {
  // TODOS_CHANGE_FORM_VALUE,
  // TODOS_CLEAR_FORM,
  // TODOS_DELETED_ONE,
  // TODOS_RESET_EDITING,
  TODOS_GET_SINGLE,
  TODOS_GOT_SINGLE,
  TODOS_GOT,
  TODOS_GET_MANY,
} from '../todos/actionTypes'

function* getTodos() {
  try {
    const data = getDataFromResult( yield TodosApi.getTodos() )
    yield put( createAction(TODOS_GOT, { data }) )
  } catch(e) {
    defaultErrorCatcher(e)
  }
}

function* getSingleTodo(id) {
  try {
    const data = getDataFromResult( yield TodosApi.getTodoSingle() )
    yield put()
  } catch(e) {

  }
}

function* todos() {
  yield takeLatest(TODOS_GET_MANY, getTodos)
  yield takeLatest(TODOS_GET_SINGLE, getSingleTodo)
}

export default function* rootSaga() {
  yield all([
    todos(),
  ])
}
