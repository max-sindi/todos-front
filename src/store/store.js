import { createStore, applyMiddleware, compose } from 'redux'
import thunk from 'redux-thunk'
import reducers from './reducers'
import { createLogger } from 'redux-logger'

const logger = createLogger({
  diff: true,
  collapsed: true,
})

const middlewares = [
  thunk,
  logger,
]

const reduxDevtools = window.__REDUX_DEVTOOLS_EXTENSION__

const store = createStore(
  reducers,
  undefined,
  compose(
    applyMiddleware(...middlewares),
    reduxDevtools ? reduxDevtools() : f => f
  )
)

export default store
