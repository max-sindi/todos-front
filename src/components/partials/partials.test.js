import React from 'react'
import { Search } from './'
import renderer from 'react-test-renderer'
import {render, fireEvent, cleanup, waitForElement} from 'react-testing-library'
import store from 'store/store'
import { Provider } from 'react-redux'

afterEach(cleanup)

test('Search renders correctly', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Search />
    </Provider>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})
