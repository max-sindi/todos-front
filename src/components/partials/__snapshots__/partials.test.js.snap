// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  className="styled__Relative-sc-186tyjq-2 gRGrvy"
>
  <input
    onChange={[Function]}
    style={
      Object {
        "width": "100%",
      }
    }
    value=""
  />
  <button
    className="styled__Button-sc-186tyjq-0 styled__DeleteButtonStyled-sc-186tyjq-1 liuCwo"
    onClick={[Function]}
  >
    <span>
      âœ•
    </span>
     
  </button>
</div>
`;
